server/api/update_user_id.py

```diff
@@ -0,0 +1,37 @@
+from flask import Blueprint, app, jsonify, request
+from bson import ObjectId
+import config
+from models.user import User
+
+update_user_id_blueprint = Blueprint('update_user_id_blueprint', __name__)
+
+
+@update_user_id_blueprint.route('/user/<user_id>', methods=["POST"])
+def update_user(user_id):
+    """
+    Update a user object in database.
+    This function expects user id as url parameter.
+    This fucntion aslo expects new values for user object 
+    in the request body. So request should use POST method and 
+    It should have body.
+    """
+    output = dict()
+    status = 400
+    data = request.json
+    if not user_id:
+        output["error"] = "Missing user id"
+    if not data:
+        output["error"] = "Missing updated data"
+    else:
```

Comment:
This structure seems a bit weird to me. I think it makes more sense to just return the error status right here instead of waiting to the end. Because for example if you're missing a userid, but somehow you have data, you end up in the `else` tree and you might encounter an exception and not realize that that was your error because you overwrite output.
I think something like this makes sense:
```
    if not user_id:
       return jsonify({"error":"Missing user id"}), status
    if not data:
        return jsonify({"error":"Missing updated data"}), status

        try:
            query = {"_id": ObjectId(user_id)}
            newValues = {"$set": data}
            result = User.update(query, newValues)
            if result:
                output["success"] = f"User with id {user_id} update suceesfully"
            else:
                output["error"] = "Error occurred while updaing"

           return jsonify(output), status
```

Reply:
Thanks Sina, you are right, If statements are not consistent. I made changes.